modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rule-builder/editor/Editor.es.js

```diff
@@ -0,0 +1,833 @@
+							onValidator={() => {}}
+						/>
+					);
+
+					await waitForElement(() => {
+						return document.querySelector('.option-selected');
+					});
+
+					const fieldLeft = document
+						.querySelectorAll('.timeline-item')[1]
+						.querySelectorAll('.ddm-field')[0]
+						.querySelector('.option-selected');
+
+					fireEvent.click(fieldLeft);
+
+					await waitForElement(() => {
+						return document.querySelector('.dropdown-menu.show');
+					});
+
+					fireEvent.click(getByText('text'));
+
+					const operator = document
+						.querySelectorAll('.timeline-item')[1]
+						.querySelectorAll('.ddm-field')[1]
+						.querySelector('.option-selected');
+
+					fireEvent.click(operator);
+
+					fireEvent.click(getByText('Is equal to'));
+
+					const actionsType = document
+						.querySelectorAll('.timeline-item')[1]
+						.querySelectorAll('.ddm-field')[2]
+						.querySelector('.option-selected');
+
+					fireEvent.click(actionsType);
+
+					fireEvent.click(getByText('other-field'));
+
+					await waitForElement(() => {
+						return document.querySelector('.ddm-field');
+					});
+
+					const fieldRight = document
+						.querySelectorAll('.timeline-item')[1]
+						.querySelectorAll('.ddm-field')[3]
+						.querySelector('.option-selected');
+
+					fireEvent.click(fieldRight);
+
+					const allFields = STRING_DATATYPE_FIELDS.concat(
+						NUMBER_TYPE_FIELDS
+					).concat(UPLOAD_TYPE_FIELD);
+
+					allFields.forEach((field) => {
+						const fieldOption = Array.from(
+							document.querySelectorAll(
+								'.dropdown-menu.show button'
+							)
+						).find((el) => el.textContent === field);
+						expect(fieldOption).toBeTruthy();
+					});
+				});
+			});
+
+			describe('Conditions logical operatores', () => {
+				it('shows the OR/AND select disabled by default', async () => {
+					const props = defaultProps();
+					render(
+						<Editor
+							{...props}
+							onChange={() => {}}
+							onValidator={() => {}}
+						/>
+					);
+
+					await waitForElement(() => {
+						return document.querySelector('.option-selected');
+					});
+
+					expect(
+						document.querySelector(
+							'.timeline-first .dropdown-toggle'
+						).disabled
+					).toBe(true);
+				});
+
+				it('enables the OR/AND select when there are more than one condition', async () => {
+					const props = defaultProps();
+					render(
+						<Editor
+							{...props}
+							onChange={() => {}}
+							onValidator={() => {}}
+							rule={newRule}
+						/>
+					);
+
+					await waitForElement(() => {
+						return document.querySelector('.option-selected');
+					});
+
+					expect(
+						document.querySelector(
+							'.timeline-first .dropdown-toggle'
+						).disabled
+					).toBe(false);
+				});
+			});
+
+			describe('Add/Remove conditions', () => {
+				it('shows the container trash when there are more than one condition', () => {
+					const props = defaultProps();
+					render(
+						<Editor
+							{...props}
```

Comment:
To avoid declaring a new variable, you can directly call the function and deconstruct the Editor, taking advantage of some micro-optimizations of the engine.

- The engine will not need to declare the variable and invoke the function and then do the props destructure, when you do `{...defaultProps()}` the engine will do this operation all at once without having to cache it for example.

```suggestion
					render(
						<Editor
							{...defaultProps()}
```